kura/org.eclipse.kura.wire.provider/src/main/java/org/eclipse/kura/internal/wire/WireServiceImpl.java

```diff
@@ -52,126 +57,167 @@
+        logger.debug(message.deactivatingWireService());
+        this.wireComponentServiceTracker.close();
+        for (final WireConfiguration wireConfiguration : this.wireConfigs) {
+            deleteWireConfiguration(wireConfiguration);
         }
+        logger.debug(message.deactivatingWireServiceDone());
     }
 
     /** {@inheritDoc} */
     @Override
     public WireConfiguration createWireConfiguration(final String emitterPid, final String receiverPid)
             throws KuraException {
-        requireNonNull(emitterPid, s_message.emitterPidNonNull());
-        requireNonNull(receiverPid, s_message.receiverPidNonNull());
+        requireNonNull(emitterPid, message.emitterPidNonNull());
+        requireNonNull(receiverPid, message.receiverPidNonNull());
 
-        s_logger.info(s_message.creatingWire(emitterPid, receiverPid));
+        logger.info(message.creatingWire(emitterPid, receiverPid));
         WireConfiguration conf = null;
         if (!emitterPid.equals(receiverPid)) {
             final String emitterServicePid = this.wireHelperService.getServicePid(emitterPid);
             final String receiverServicePid = this.wireHelperService.getServicePid(receiverPid);
-            if ((emitterServicePid == null) || (receiverServicePid == null)) {
-                throw new KuraException(KuraErrorCode.INTERNAL_ERROR, s_message.componentPidsNull());
+            if (emitterServicePid == null || receiverServicePid == null) {
+                throw new KuraException(KuraErrorCode.CONFIGURATION_ERROR, message.componentPidsNull());
             }
             if (!(this.wireHelperService.isEmitter(emitterPid) || this.wireHelperService.isReceiver(receiverPid))) {
-                throw new KuraException(KuraErrorCode.INTERNAL_ERROR, s_message.componentsNotApplicable());
+                throw new KuraException(KuraErrorCode.CONFIGURATION_ERROR, message.componentsNotApplicable());
             }
             conf = new WireConfiguration(emitterPid, receiverPid, null);
             final Wire wire = this.wireAdmin.createWire(emitterServicePid, receiverServicePid, null);
             if (wire != null) {
                 conf.setWire(wire);
                 this.wireConfigs.add(conf);
             }
-            s_logger.info(s_message.creatingWireDone(emitterPid, receiverPid));
+            logger.info(message.creatingWireDone(emitterPid, receiverPid));
         }
         return conf;
     }
 
     /**
      * Create the wires based on the provided wire configurations
      *
-     * @throws KuraException
-     *             if there doesn't exist any Wire Component having provided
-     *             emitter PID or any Wire Component having provided receiver
-     *             PID
      */
-    synchronized void createWires() throws KuraException {
-        s_logger.debug(s_message.creatingWires());
+    synchronized void createWires() {  // TODO: this needs to be refactored
```

Comment:
Timing issues. All the todos are tracked and will be solved in M2. For now these methods are taken as they are.